docs/concept-workflows.md

```diff
@@ -60,70 +240,45 @@ public class SubscriptionWorkflowImpl implements SubscriptionWorkflow {
-- `RejectDuplicate` means that it is not allowed to start a Workflow execution using the same Workflow Id at all.
+Again, it is important to note that this code directly implements the business logic, and if any of the invoked operations (aka [Activities](/docs/concept-activities)) take a long time, the code is not going to change.
 
-The default is `AllowDuplicate`.
+It is completely okay to be blocked on `chargeMonthlyFee` for a day or more if the downstream processing service is down or not responding.
+In the same way it is a completely normal operation to sleep for 30 days directly inside the Workflow code.
 
-To distinguish multiple runs of a Workflow with the same Workflow Id, Temporal identifies a Workflow with two Ids: `Workflow Id` and `Run Id`. `Run Id` is a service-assigned UUID. To be precise, any Workflow is uniquely identified by a triple: `Namespace`, `Workflow Id` and `Run Id`.
+The Temporal Server has practically no scalability limits on the number of open Workflow instances, so this code can be used over and over even if your application has hundreds of millions of customers.
 
-## Child Workflow
+## FAQ
 
-A Workflow can execute other Workflows as `child Workflows`. A child Workflow completion or failure is reported to its parent.
+**Is there a limit to length of time a Workflows can run?**
 
-Some reasons to use child Workflows are:
+Workflows intended to run indefinitely should be written with some care.
+Temporal stores the complete event history for the entire life of a Workflow Execution.
+There is a maximum limit of 50,000 events that is enforced by the Server, and you should try to avoid getting close to this limit; The Temporal Server puts out a warning at every 10,000 events.
 
-- A child Workflow can be hosted by a separate set of workers which don't contain the parent Workflow code. So it would act as a separate service that can be invoked from multiple other Workflows.
-- A single Workflow has a limited size. For example, it cannot execute 100k Activities. Child Workflows can be used to partition the problem into smaller chunks. One parent with 1000 children each executing 1000 Activities is 1 million executed Activities.
-- A child Workflow can be used to manage some resource using its Id to guarantee uniqueness. For example, a Workflow that manages host upgrades can have a child Workflow per host (host name being a Workflow Id) and use them to ensure that all operations on the host are serialized.
-- A child Workflow can be used to execute some periodic logic without blowing up the parent history size. When a parent starts a child, it executes periodic logic calling that continues as many times as needed, then completes. From the parent point of view, it is just a single child Workflow invocation.
+The idiomatic way to handle indefinitely running Workflows is to use the "Continue-as-new" feature, which is available in all SDKs.
```

Comment:
Do we have a task to document continue-as-new? I don't think we have a page that talks about it as a concept.

Reply:
i tried to answer it a bit in the "FAQ" section https://docs.temporal.io/docs/concept-workflows/#faq since we talk about it to every customer, but the api documentation itself is spread across the Go/Java SDK docs